client/protocols/guidance.py

```diff
@@ -0,0 +1,178 @@
+from client.protocols.common import models
+from client.utils import auth
+from urllib.request import urlopen
+import json
+import os
+import sys
+
+TGSERVER = "http://tg-server.app.cs61a.org/"
+LOCAL_TG = "tests/tg.ok_tg"
+OK_GUIDANCE = "tests/.ok_guidance"
+
+lambda_string_key_to_func = {
+    'none': lambda info, strMisU: None, 
+    'ki': lambda info, strMisU: info['ki'],
+    'reteach': lambda info, strMisU: info['dictMisU2ReteachMsg'].get(strMisU)
+}
+  
+# If student forces guidance messages to show, we will assign treatment group number below
+guidance_flag_dict = [1]
+
+class GuidanceProtocol(models.Protocol):
```

Comment:
This doesn't look like it should be a protocol. Protocols are designed to run sequentially in the main [ok.py](https://github.com/Cal-CS-61A-Staff/ok-client/blob/master/client/cli/ok.py#L130) script, and protocols shouldn't be calling other protocols. If this is in fact supposed to be a protocol, you need to override the methods found in [client/protocols/common/models.py](https://github.com/Cal-CS-61A-Staff/ok-client/blob/master/client/protocols/common/models.py).

I suggest making this some sort of utility class instead and not inheriting from models.Protocol.